modules/adnuntiusBidAdapter.js

```diff
@@ -154,28 +154,66 @@ const storageTool = (function () {
     storage.setDataInLocalStorage(METADATA_KEY, JSON.stringify(metaDataForSaving));
   };
 
-  const getUsi = function (meta, ortb2, bidParams) {
-    // Fetch user id from parameters.
-    for (let i = 0; i < bidParams.length; i++) {
-      const bidParam = bidParams[i];
-      if (bidParam.userId) {
-        return bidParam.userId;
-      }
+  const getFirstValidValueFromArray = function(arr, param) {
+    const example = (arr || []).find((b) => {
+      return deepAccess(b, param);
+    });
+    return example ? deepAccess(example, param) : undefined;
+  };
+
+  const normaliseLegacyIds = function(obj) {
+    const result = {};
+    if (!isPlainObjectWithKeys(obj)) {
+      return result;
     }
-    if (ortb2 && ortb2.user && ortb2.user.id) {
-      return ortb2.user.id
+    for (const [key, value] of Object.entries(obj)) {
+      if (isPlainObjectWithKeys(value)) {
+        const theId = value[Object.keys(value)[0]];
+        if (isStr(theId)) {
+          result[key] = theId;
+        }
+      } else if (isStr(value)) {
+        result[key] = value;
+      }
     }
-    return (meta && meta.usi) ? meta.usi : false
-  }
+
+    return result;
+  };
 
   return {
-    refreshStorage: function (bidderRequest) {
-      const ortb2 = bidderRequest.ortb2 || {};
+    refreshStorage: function (validBidRequests, bidderRequest) {
       const bidParams = (bidderRequest.bids || []).map((b) => {
         return b.params ? b.params : {};
       });
       metaInternal = getMetaDataFromLocalStorage(bidParams).reduce((a, entry) => ({ ...a, [entry.key]: entry.value }), {});
-      metaInternal.usi = getUsi(metaInternal, ortb2, bidParams);
+      const bidParamUserId = getFirstValidValueFromArray(bidParams, 'userId');
+      const ortb2 = bidderRequest.ortb2 || {};
+
+      if (isStr(bidParamUserId)) {
+        metaInternal.usi = bidParamUserId;
+      } else if (isStr(ortb2?.user?.id)) {
+        metaInternal.usi = ortb2.user.id;
+      } else {
+        const unvettedOrtb2Eids = deepAccess(ortb2, 'user.ext.eids');
```

Comment:
the rest of this feels like a mess, this should be all you need. What's all the additional processing for?

Reply:
The additional processing was for fallbacks.

My understanding was that if `ortb2.user.ext.eids` isn't defined, you can check userIdAsEids. And then to support legacy ortb implementations, you can look into `userId` and turn that into `eids`.

But if all that's needed is `ortb2.user.ext.eids`, then that's fine and I can remove all the rest of the code as I have just done.